### PR TITLE
edited branch manager handbook

### PR DESCRIPTION
**Pull request by jimangel**
#### What type of PR is this:

/kind documentation

#### What this PR does / why we need it:

General edits as I went deeper into the branch manager handbook. Open to feedback / edits.

cc: @kubernetes/release-engineering

### CODE REVIEW

File: release-engineering/role-handbooks/branch-manager.md [hunk 1]
```diff
@@ -73,38 +72,45 @@ In this handbook, we will make several references to Kubernetes releases, milest
...
+| "next release", "next milestone" | `x.y+1` | Kubernetes 1.19 |
+| "release no longer in support" | `x.y-4` | Kubernetes 1.14 |
 
 **As an editor of this content, Branch Managers should periodically update these conventions and the examples contained within this handbook.**
 
+### Minimum Requirements
+
+- Familiarity with basic Unix commands and able to debug shell scripts.
+- Familiarity with branched source code workflows via `git` and associated `git` command line invocations.
+- General knowledge of Google Cloud (Cloud Build and Cloud Storage).
```
Comment (tpepper) [inline release-engineering/role-handbooks/branch-manager.md:95]: Could we lower the bar a bit by being more generic?  Eg:

- Abstract knowledge of public cloud provider patterns (eg: Amazon, Google, Microsoft, VMware, etc.), building code from containerize build environments, and cloud storage, with the ability to explore and map that knowledge to specifics on GCP (Cloud Build and Cloud Storage).

I think a lot of us didn't have for example GCB specific knowledge until we got into the role.  We should rather have specializing generalists who are able to work in the abstracted problem space versus people who can or will only solve a problem if it is being done on one specific provider.

File: release-engineering/role-handbooks/branch-manager.md [hunk 2]
```diff
@@ -73,38 +72,45 @@ In this handbook, we will make several references to Kubernetes releases, milest
...
-6. Select Associates and guide them in preparation to become the section lead for the next cycle.
-7. Delegate tasks to the Associates (where applicable), so that they may exercise the release process.
-8. Update this handbook where appropriate for the next release cycle.
-9. Have willingness to accommodate with different timezone esp. for the release team and your Associates.
-10. Participate in conversations that happen on [#sig-release] and [#release-management]
+The release branch manager is responsible for cutting a version of [Kubernetes]. Each release is a three month cycle where as branch manager it's expected you:
+
+1. Cut releases for the next release cycle as specified on the timeline in `sig-release/releases/release-x.y/README.md`
+   - For example, during the v1.16 cycle you're expected to allocate time to cut releases on the dates detailed in [sig-release/releases/release-1.16/README.md][release-1.16]. The bulk of your time commitment is during release cut days.
+1. Allocate time to setup your system to run the release tools; eventually you'll become more familiar with the process.
```
Comment (tpepper) [inline release-engineering/role-handbooks/branch-manager.md:105]: Maybe a mention here that this is ongoing.  Code change in the tooling drive change requirements in the host runtime system.  Reads as if it is a one-time up front cost.

**Comment by justaugustus** [comment]
Thanks for the cleanups, @jimangel!

> A lot of really good cleanups here. Need to sync with @justaugustus who was about to post a big rework of this document.

w.r.t. ^^ @kubernetes/release-engineering -- please hold on any future planned edits to these handbooks, as I'm planning a rework.

/lgtm
/approve

**Pull request merged**